### ISSUE TITLE
loading luarock module "regkex" failed: undefined symbol: lua_newuserdatauv

### ISSUE THREAD

**Issue by qrockz** [mfussenegger/nlua#5]
Hi,

I've installed **_regkex_**
`sudo luarocks install regkex --lua-version=5.4`

**mycode.lua**
```
package.cpath = package.cpath .. ";/usr/local/lib/lua/5.4/?.so"
local Regex = require("regkex")
```
Error :(
```
cat reg.lua | nlua.lua
E5113: Error while calling lua chunk: error loading module 'regkex' from file '/usr/local/lib/lua/5.4/regkex.so':
	/usr/local/lib/lua/5.4/regkex.so: undefined symbol: lua_newuserdatauv
```
```
nlua.lua -v
NVIM v0.9.2
Build type: Release
LuaJIT 2.1.1692716794
system vimrc file: "$VIM/sysinit.vim"
fall-back for $VIM: "/__w/neovim/neovim/build/nvim.AppDir/usr/share/nvim"

Run :checkhealth for more info
```

If I use the regular lua interpreter _**mycode.lua**_ works.

```
lua -v
Lua 5.4.6 
```

**Comment by mfussenegger** [comment]
`lua_newuserdatauv` is not available in Lua 5.1

You'll have to install regkex using a 5.1 interpreter. Either regular one, or via a custom config as documented in https://github.com/mfussenegger/nlua#luarocks-setup should work.

**Comment by qrockz** [comment]
`sudo luarocks install regkex --lua-version=5.1`
_--lua-version=_**5.1** has worked 4 me.


> 
> 
> [...] or via a **custom config** as documented in https://github.com/mfussenegger/nlua#luarocks-setup should work.

EDIT: 
Hmm, so I should have run `nluarocks install regexk` in the first place 
instead of `luarocks install regexk  --lua-version=5.1`
because _config-nlua.lua_ has already configured  _lua-version=5.1_
thus `nluarocks install <whatever rocks>` automatically will be installed by the right interpreter version 5.1 which nvim-lua is compatible with.
